validators/cgroup_validator_linux.go

```diff
@@ -44,28 +44,117 @@ func (c *CgroupsValidator) Name() string {
 }
 
 const (
-	cgroupsConfigPrefix = "CGROUPS_"
-	unifiedMountpoint   = "/sys/fs/cgroup"
+	cgroupsConfigPrefix      = "CGROUPS_"
+	defaultUnifiedMountpoint = "/sys/fs/cgroup"
+	mountsFilePath           = "/proc/mounts"
+	mountInfoPath            = "/proc/self/mountinfo"
 )
 
+// getUnifiedMountpoint checks if the default mount point is available.
+// If not, it parses the mounts file to find a valid cgroup mount point.
+func getUnifiedMountpoint() (string, error, error) {
+	isCgroupfs, err := isMountPointCgroupfs(defaultUnifiedMountpoint)
+	if isCgroupfs {
+		return defaultUnifiedMountpoint, err, nil
+	}
+	mountPath, err := parseMountsFile(mountsFilePath)
+	return mountPath, err, fmt.Errorf("%q is not cgroupfs: %v", defaultUnifiedMountpoint, err)
+}
+
+func isMountPointCgroupfs(path string) (bool, error) {
+	f, err := os.Open(mountInfoPath)
+	if err != nil {
+		return false, err
+	}
+
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !strings.Contains(line, "cgroup") {
+			continue
+		}
+		// example fields: `28 22 0:25 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime shared:4 - cgroup2 cgroup2 rw,seclabel`
+		fields := strings.Fields(line)
+		if len(fields) >= 9 {
+			if fields[5] == path {
+				i := 6
+				for i < len(fields) && fields[i] != "-" {
+					i++
+				}
+				if len(fields)-i <= 3 {
+					return false, fmt.Errorf("expect 3 fields after `-` in mount info file %q", path)
```

Comment:
```suggestion
					return false, fmt.Errorf("in mount info file %q, expected 3 fields after `-`, but got %d", path, xxxx)
```

also why can this happen?